### ISSUE TITLE
JavaScript heap out of memory

### ISSUE THREAD

**Issue by RuggeriLuca** [postmanlabs/newman#2229]
Hello,
We have the below case :
1 - Create a collection on postman using Pre-Request Script and Tests scripts .
2 - Export the collection
3 - use the below command line to run the collection for 100 iteration :slight_smile:

newman run C:\temp\004_L10_Orchestrator_L10_UAT_V4.0.postman_collection.json -g C:\temp\Euro_globals.json -e C:\temp\UAT-L10-WagerOperations.postman_environment.json --folder “5104 - Tzoker (Channel 2)” -k -n 100

after 50 iteration the run crash with a nodejs Heap Memory Error .

How we can fix it ?

This is not a performance test because each iteration have different data

<— Last few GCs —>

[10028:000001BE90168DD0] 355619 ms: Scavenge 2042.2 (2049.5) -> 2041.1 (2049.5) MB, 4.4 / 0.0 ms (average mu = 0.234, current mu = 0.129) allocation failure
[10028:000001BE90168DD0] 355636 ms: Scavenge 2042.7 (2049.7) -> 2041.9 (2050.7) MB, 5.8 / 0.0 ms (average mu = 0.234, current mu = 0.129) allocation failure
[10028:000001BE90168DD0] 355660 ms: Scavenge 2044.7 (2052.0) -> 2043.9 (2056.1) MB, 5.7 / 0.0 ms (average mu = 0.234, current mu = 0.129) allocation failure

<— JS stacktrace —>

==== JS stack trace =========================================

0: ExitFrame [pc: 00007FF788EA154D]
1: StubFrame [pc: 00007FF788F11F49]
Security context: 0x00d1dc3c0921
2: baseClone [0000008D72BE2A39] [C:\Users\Agent\AppData\Roaming\npm\node_modules\newman\node_modules\lodash\lodash.js:1] [bytecode=000001855E0FE891 offset=0](this=0x0098e94c2531 ,0x011436a82d41 <String[#7]: integer>,5,0x0110437ab189 <JSFunction (sfi = 00000150CA97CAB1)>,0x011436a83259 <String[#4]…

FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory

Writing Node.js report to file: report.20200122.122008.10028.0.001.json
Node.js report completed
1: 00007FF788290BEF napi_wrap+126287
2: 00007FF78822DF26 public: bool __cdecl v8::base::CPU::has_sse(void)const __ptr64+34950
3: 00007FF78822EBF6 public: bool __cdecl v8::base::CPU::has_sse(void)const __ptr64+38230
4: 00007FF788A57FEE private: void __cdecl v8::Isolate::ReportExternalAllocationLimitReached(void) __ptr64+94
5: 00007FF788A3F741 public: void __cdecl v8::SharedArrayBuffer::Externalize(class std::shared_ptr const & __ptr64) __ptr64+785
6: 00007FF7889067AC public: static void __cdecl v8::internal::Heap::EphemeronKeyWriteBarrierFromCode(unsigned __int64,unsigned __int64,class v8::internal::Isolate * __ptr64)+1468
7: 00007FF788911AF0 public: void __cdecl v8::internal::Heap::ProtectUnprotectedMemoryChunks(void) __ptr64+1312
8: 00007FF78890E5E4 public: static bool __cdecl v8::internal::Heap::PageFlagsAreConsistent(class v8::internal::HeapObject)+3188
9: 00007FF788903CF3 public: bool __cdecl v8::internal::Heap::CollectGarbage(enum v8::internal::AllocationSpace,enum v8::internal::GarbageCollectionReason,enum v8::GCCallbackFlags) __ptr64+1283
10: 00007FF788902484 public: void * __ptr64 __cdecl v8::internal::Heap::AllocateExternalBackingStore(class std::function<void * __ptr64 __cdecl(unsigned __int64)> const & __ptr64,unsigned __int64) __ptr64+1396
11: 00007FF788924554 public: class v8::internal::Handle __cdecl v8::internal::factory::NewFillerObject(int,bool,enum v8::internal::AllocationType,enum v8::internal::AllocationOrigin) __ptr64+196
12: 00007FF78867E841 ??4iterator@JumpTableTargetOffsets@interpreter@internal@v8@@QEAAAEAV01234@$$QEAV01234@@Z+1681
13: 00007FF788EA154D public: virtual bool __cdecl v8::internal::SetupIsolateDelegate::SetupHeap(class v8::internal::Heap * __ptr64) __ptr64+517453
14: 00007FF788F11F49 public: virtual bool __cdecl v8::internal::SetupIsolateDelegate::SetupHeap(class v8::internal::Heap * __ptr64) __ptr64+978761
15: 00007FF788E2CE6B public: virtual bool __cdecl v8::internal::SetupIsolateDelegate::SetupHeap(class v8::internal::Heap * __ptr64) __ptr64+40555
16: 00007FF788E2CE6B public: virtual bool __cdecl v8::internal::SetupIsolateDelegate::SetupHeap(class v8::internal::Heap * __ptr64) __ptr64+40555
17: 00007FF788E26A69 public: virtual bool __cdecl v8::internal::SetupIsolateDelegate::SetupHeap(class v8::internal::Heap * __ptr64) __ptr64+14953
18: 000003F9E5B1B13E

**To Reproduce**
1- Create A collection 
2- run 200 iteration using newman

**Comment by shamasis** [comment]
@RuggeriLuca @codenirvana ... can this be temporarily bypassed using max old space node memory limit ceiling increasing parameters? 

In parallel, we should add a no-summary option (since summary for reporter is the only thing that aggregates memory) or a disk backed summary method?

**Comment by pawanakhil** [comment]
Try running this script before running the collection, so that you can allow more memory
export NODE_OPTIONS=--max_old_space_size=8192

Hope this helps

**Comment by devilmind81** [comment]
Hi all,
I've installed the newman version 5.0.0 and I'm running the newman in a docker container so I want manage it with the minimum Memory configuration.
I'm running a test using JMETER passing a collection with only one post.
I've observed that  with only 35 concurrency calls the memory goes over 2 GB and it is going in JavaScript heap out of memory.

How it possible to avoid this memory increment? 

Thanks
